SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -1532,6 +1532,11 @@ void ActorSystem::RemoveActor(const Worker_EntityId EntityId)
 		return;
 	}
 
+	if (Actor->HasAuthority())
+	{
+		Actor->OnAuthorityLost();
```

Comment:
I don't think we should be calling callbacks on deleted actors.
It means we are giving users callbacks while we've only applied partial state to things.

Reply:
Why does this fix the issue outlined in the doc?

Reply:
We should keep a certain order of OnAuthorityLost and delete the actor.
I worried about if the end-user has some logic base on the order.

If they want to do something A in the OnAuthorityLost function and something B else in the removal callback. 
But A will never be called in this situation.
It's valuable to have OnAuthorityLost here.

Reply:
@jacquese Have the explanation in my doc: Why call OnAuthorityLost before DeleteActor?
Pleased to have more info for you if you want.

Reply:
I posted a comment in the doc, but should the fix be to change how the events get processed to such that component events will be processed before entities are removed? My general concern is whether it's the viewdelta code which needs to be changed?

Reply:
I think I follow now. Potentially, instead of including only the snippets of authority lost code that we know are causing issues, we should instead maintain the component sets we have authority over and call authority lost inside `EntityRemoved`, before calling `RemoveActor`, the inverse of what happens in `EntityAdded`. This would mean the exact same flow is used for implicit authority losts during an entity removal as a normal authority lost.

The `ActorSystem` treats any remove entity as a "clean up everything to do with this entity" rather than simply a remove entity, as it means the entity is no longer an Actor by this worker. To that end processing the auth losts for the entity makes sense here if they wouldn't otherwise, especially as we do the opposite for entity added.

Reply:
Yep, that makes sense, I think we should go with that.

Reply:
I don't know what changing the view delta code would mean except to remove information about what components / authority were present before (which I generally found not having it was a pain during prototyping). It doesn't contain any ordering information by design and doesn't need to. Unless you mean the code reading the view delta is doing things in the wrong order.

My issue with processing events on removed entities is when in the "setting the view" stage of the tick does the event get called? Do you know what else has been removed? Do you falsely think that you still have authority over something else at this point and are able to write to it. As you aren't able to write to any other entity during this time is the only thing you should be doing resource clean up?
I think that it might be reasonable to have such a callback in this place but I don't like adding that as a driveby to another pr without prior discussion.